spec/defines/value_spec.rb

```diff
@@ -38,4 +38,55 @@
 
     it { is_expected.to compile.and_raise_error(/parameter 'type'/) }
   end
-end
\ No newline at end of file
+
+  context 'Given a resource with string data' do
+    let(:params) {{
+       :key  => 'HKLM\Software\Vendor',
+       :data => 'KingKongBundy'
+    }}
+
+    it { is_expected.to compile }
+  end
+
+  ['dword', 'qword'].each do |numeric|
```

Comment:
The tests here cover all the valid types per https://github.com/puppetlabs/puppetlabs-registry#type - `string`, `array`, `dword`, `qword`, `binary` and `expand`